wrappers/nodejs/src/addon.cpp

```diff
@@ -1455,47 +1455,89 @@ Nan::Persistent<v8::Function> RSSyncer::constructor_;
 
 class Options {
  public:
-  explicit Options(rs2_options* options) : options_(options), error_(nullptr) {}
+  explicit Options(rs2_options* options = nullptr) : options_(options),
+      error_(nullptr) {}
 
-  ~Options() {
+  virtual ~Options() {
     if (error_) rs2_free_error(error_);
   }
 
-  bool SupportsOption(rs2_option option) {
-    return (rs2_supports_option(options_, option, &error_) != 0);
+  void SetOptionsPointer(rs2_options* options) {
+    options_ = options;
```

Comment:
This seems very dangerous will cause point dangling

Reply:
Yes, I can remove this, in fact, this pointer can be retrieved by get it from the object: the rs2_sensor* of RSSensor or the rs2_processing_block* of RSColorizer.